craft_parts/lifecycle_manager.py

```diff
@@ -103,6 +103,40 @@ def project_info(self) -> ProjectInfo:
         """Obtain information about this project."""
         return self._project_info
 
+    def clean(
+        self, step: Optional[Step] = None, *, part_names: List[str] = None
+    ) -> None:
+        """Clean the specified step and parts.
+
+        Cleaning a step removes its state and all artifacts generated in that
+        step and subsequent steps for the specified parts.
+
+        :para step: The step to clean.
+        :param part_names: The list of part names to clean. If not specified,
+            all parts will be cleaned and work directories will be removed.
+        """
+        if not step:
+            step = Step.PULL
```

Comment:
How weird would be to put this default in the signature above?

Reply:
Good idea, will do that.